eox_nelp/openedx_filters/tests/xapi/tests_filters.py

```diff
@@ -205,16 +199,33 @@ def tearDown(self):
         modulestore.reset_mock()
         self.transformer.reset_mock()
 
-    @data(None, "")
+    def set_activity(self, definition_type):
+        """Set the activity argument for the given type"""
+        self.activity = Activity(
+            id="https://example.com/xblock/block-v1:edx+CS105+2023-T3+type@problem+block@0221040b086c4618b6b2b2a554558",
+            definition=ActivityDefinition(
+                type=definition_type,
+                name=LanguageMap(en="old-testing-course"),
+            )
+        )
+
+    @data(
+        (constants.XAPI_ACTIVITY_MODULE, None),
+        (constants.XAPI_ACTIVITY_MODULE, ""),
+        (constants.XAPI_ACTIVITY_QUESTION, None),
+        (constants.XAPI_ACTIVITY_QUESTION, ""),
+    )
+    @unpack
```

Comment:
Why did you choose to use the unpack instead of using an object like a dict. And inside the test use the dict to get the keys related. eg `definiton_type`?

Reply:
As you have described that implementation requires more changes and more code